### PR TITLE
add new external link filter

### LINKED ISSUE #158: Task cards on benchmarks page should not open in new tab in deploy preview
We don't know how to fix this and it doesn't affect the main site, but it's still a pain for previews.

### PR DESCRIPTION
**Pull request by KaiWaldrant**
## Describe your changes
Use a custom filter to determine if the link is external. The filter is very generic which makes it possible to keep the `site-url: https://openproblems.bio`

## Issue ticket number and link
Closes #158

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [x] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [x] CI Preview succeeds and looks good!

### CODE REVIEW

**Comment by KaiWaldrant** [comment]
It is fixed for the benchmark cards but the bibliography links are not fixed

**Comment by KaiWaldrant** [comment]
the biblio ref links are opening in a new tab on the released website too #180

**Comment by KaiWaldrant** [comment]
#180 is intentional so the PR can be reviewed

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 * Dark mode was added.
 
 * Changed the bibliography hero.
+  
+* use filter to determine external link
```
Comment (rcannood) [inline CHANGELOG.md:12]: Can you document this as a bug fix?

```suggestion
## BUG FIXES

* Fix internal links being treated as external links in PR previews (#179).
```

**Review by rcannood — CHANGES_REQUESTED** [review]

**Comment by rcannood** [comment]
Is it possible `link-external-newwindow: true` shouldn't be removed? None of the links on the main page have target blank.

**Review by rcannood — COMMENTED** [review]
Is it possible `link-external-newwindow: true` shouldn't be removed? None of the links on the main page have target blank.

**Comment by KaiWaldrant** [comment]
you are totally correct. forgot to test the external links ofcourse ...
I also added the external link icon.

**Comment by KaiWaldrant** [comment]
hmm the results card are not fixed now :|

**Comment by scottgigante-immunai** [comment]
Do we want to keep `link-external-icon: true`? I could go either way on this. Also, looks like navbar docs links still don't open in new tab.

**Comment by KaiWaldrant** [comment]
the alternative regex seems to fit the bill.
Wanted to add the icons to see what it looked like. I'm either way, maybe it is not necessary as the links will open in a new window anyway.

**Comment by rcannood** [comment]
> Also, looks like navbar docs links still don't open in new tab.

It's really strange that the links in the navbar indeed don't open in a new tab.

**Comment by rcannood** [comment]
Looks like this is a feature that was added to quarto 1.3. I opened a separate PR for this (#188).

**Review by rcannood — APPROVED** [review]

**Pull request merged**